diff --git a/zed_wrapper/launch/zed_camera.launch.py b/zed_wrapper/launch/zed_camera.launch.py
index f5a1f0a..66f7a71 100644
--- a/zed_wrapper/launch/zed_camera.launch.py
+++ b/zed_wrapper/launch/zed_camera.launch.py
@@ -35,6 +35,7 @@ from launch_ros.actions import (
     LoadComposableNodes
 )
 from launch_ros.descriptions import ComposableNode
+from launch_ros.parameter_descriptions import ParameterValue
 
 # Enable colored output
 os.environ["RCUTILS_COLORIZED_OUTPUT"] = "1"
@@ -294,8 +295,8 @@ def launch_setup(context, *args, **kwargs):
                 'general.camera_model': camera_model_val,
                 'svo.svo_path': svo_path,
                 'svo.publish_svo_clock': publish_svo_clock,
-                'general.serial_number': serial_number,
-                'general.camera_id': camera_id,
+                'general.serial_number': ParameterValue(serial_number, value_type=int),
+                'general.camera_id': ParameterValue(camera_id, value_type=int),
                 'pos_tracking.publish_tf': publish_tf,
                 'pos_tracking.publish_map_tf': publish_map_tf,
                 'sensors.publish_imu_tf': publish_imu_tf,
